core/src/test/scala/io/projectglow/tertiary/LinearRegressionSuite.scala

```diff
@@ -304,6 +320,24 @@ class LinearRegressionSuite extends GlowBaseTest {
     assert(e.getCause.getMessage.contains(error))
   }
 
+  // Our linear regression algorithm projects the genotypes onto orthogonal complement of the
+  // covariate vector space. When the genotypes are a linear combination of some covariates, this
+  // projection is 0, so beta, standard error, and p value are all undefined. However, finite
+  // precision can instead cause the error to be massive with respect to beta.
+  test("negligible p value if genotypes are in covariate span") {
+    val testData = generateTestData(30, 1, 1, true, 1)
+    val genotypes = twoDArrayToBreezeMatrix(testData.covariates.toArray)(::, 1)
+    val phenotypes = new DenseVector[Double](testData.phenotypes.toArray)
+    val ctx = CovariateQRContext.computeQR(twoDArrayToSparkMatrix(testData.covariates.toArray))
+
+    assertThrows[SingularMatrixException] {
+      olsBaseline(genotypes.toArray, phenotypes.toArray, testData.covariates.toArray)
+    }
+
+    val results = LinearRegressionGwas.runRegression(genotypes, phenotypes, ctx)
+    assert(results.pValue > 0.999999 || results.pValue.isNaN)
```

Comment:
Does the p-value change?

Reply:
Yes, it can be NaN.

Reply:
I believe that NaN is actually the proper value since `qtx` should be 0, so all computations include a `0/0` term.